indico/core/db/sqlalchemy/locations.py

```diff
@@ -248,3 +248,48 @@ def location_data(self):
             return {'source': data_source, 'venue': data_source.venue, 'room': data_source.room,
                     'room_name': data_source.room_name, 'venue_name': data_source.venue_name,
                     'address': data_source.address, 'inheriting': self.inherit_location}
+
+    @property
+    def location_data_for_ui(self):
+        """All location data for the item, meant to be used in the location
+        widget.
+        """
+        return {
+            'address': self.location_data['address'],
+            'room_name': self.location_data['room_name'],
+            'room_id': self.location_data['room'].id if self.location_data['room'] else '',
+            'venue_name': self.location_data['venue_name'],
+            'venue_id': self.location_data['venue'].id if self.location_data['venue'] else '',
+        }
+
+    @location_data.setter
+    def location_data(self, data):
+        self.inherit_location = data['inheriting']
+        if self.inherit_location:
+            self.room = None
+            self.venue = None
+            self.venue_name = ''
+            self.room_name = ''
+            self.address = ''
+        else:
+            self.room = data.get('room')
+            self.venue = data.get('venue')
+            self.address = data.get('address')
+            if self.room:
+                self.venue_name = ''
```

Comment:
Wouldn't it be better to initialize `self.room_name = ''` and `self.venue_name = ''` at the top of the function and then update it like, e.g. `if not self.venue: self.venue_name = data.get('venue_name')`